src/appfl/algorithm/server_federated.py

```diff
@@ -62,18 +65,26 @@ def update(self, local_states: OrderedDict):
 
         """ residual calculation """
         super(FedServer, self).primal_residual_at_server()
-
+ 
         """ change device """
-        for i in range(self.num_clients):
-            for name, _ in self.model.named_parameters():
+        for i in range(self.num_clients): 
+            for name in self.model.state_dict():
```

Comment:
For `BatchNorm` layers, the `state_dict` also keeps track of the `num_batches_tracked` parameters
[What ‘num_batches_tracked’ in the new BN is for?](https://discuss.pytorch.org/t/what-num-batches-tracked-in-the-new-bn-is-for/27097), which is an integer and should not be updated using averaging. Do you think we should handle this param separately? 

Otherwise, this update should work well for most of the parameters.

Reply:
1. From the source code (https://github.com/pytorch/pytorch/blob/10c4b98ade8349d841518d22f19a653a939e260c/torch/nn/modules/batchnorm.py#L58-L81), ``num_batches_tracked`` is not used under the default setting, i.e., ``momentum=0.1``. 

2. If``momentum=None``, then it will use decreasing factor, 1/``num_batches_tracked``, where ``num_batches_tracked`` increases per every iteration. Do you have a reason why we should not use the averaging of ``num_batches_tracked``?